libsawtooth/src/journal/publisher/mod.rs

```diff
@@ -132,3 +131,88 @@ impl Clone for Box<dyn SyncPublisher> {
+    previous_block: BlockPair,
+    /// A reference to the previous block that's tracked by the block manager. This reference is
+    /// dropped when the candidate block is dropped (on finalization).
+    _previous_block_reference: BlockRef,
+    /// Handles execution of the transactions that will be in the block
+    scheduler: Box<dyn Scheduler>,
+    /// Verifies that the block is valid according to the block validation rules
+    validation_rule_enforcer: ValidationRuleEnforcer,
+
+    /// Tracks which scheduled batches were injected (not taken from the pending batches pool)
+    injected_batch_ids: HashSet<String>,
+    /// Tracks the IDs of all batches that have been scheduled, in the order they were scheduled
+    scheduled_batch_ids: Vec<String>,
+    /// Tracks the IDs of all transactions that have been scheduled
+    scheduled_txn_ids: HashSet<Vec<u8>>,
+
+    /// Used to receive batch execution results from the scheduler. This channel is consumed when
+    /// the block is completed (summarized/finalized) or cancelled.
+    batch_results: Receiver<Option<BatchExecutionResult>>,
+    /// Saves all batches that were executed and will be put in the block on finalization. This is
+    /// populated when the block is completed (summarized/finalized).
+    executed_batches: Vec<Batch>,
+    /// Saves the state root hash that will be put in the block on finalization. This is populated
+    /// when the block is completed (summarized/finalized).
+    state_root_hash: Option<String>,
+    /// Saves the summary of the block's batches. This is populated when the block is completed
+    /// (summarized/finalized).
+    summary: Option<Vec<u8>>,
+}
+
+impl CandidateBlock {
+    /// Creates a new candidate block
+    pub fn new(
+        batch_results: Receiver<Option<BatchExecutionResult>>,
+        max_batches_per_block: Option<usize>,
+        permission_verifier: PermissionVerifier,
+        previous_block: BlockPair,
+        previous_block_reference: BlockRef,
+        scheduler: Box<dyn Scheduler>,
+        validation_rule_enforcer: ValidationRuleEnforcer,
+    ) -> Self {
+        Self {
+            max_batches_per_block,
+            permission_verifier,
+            previous_block,
+            _previous_block_reference: previous_block_reference,
+            scheduler,
+            validation_rule_enforcer,
+
+            injected_batch_ids: Default::default(),
+            scheduled_batch_ids: Default::default(),
+            scheduled_txn_ids: Default::default(),
+
+            batch_results,
+            executed_batches: Default::default(),
+            state_root_hash: Default::default(),
+            summary: Default::default(),
+        }
+    }
+
+    /// Returns `true` if batches can still be scheduled; returns `false` otherwise.
+    ///
+    /// Batches can be scheduled for the candidate block when both of these conditions are true:
+    /// * The maximum number of batches that a block can contain have aleady been scheduled
```

Comment:
typo "aleady" -> "already". Also I think this should say "has NOT already been scheduled"

Reply:
Fixed